src/main.rs

```diff
@@ -1,7 +1,17 @@
 mod config;
+mod server;
+mod error;
 
-fn main() {
-	let config = config::Config::load();
+#[tokio::main]
+async fn main() {
+	let config = config::Config::load().unwrap();
 	println!("Loaded config:");
 	println!("{config:?}");
+
+	match server::serve(&config).await {
+		Ok(_) => {}
```

Comment:
I don't know how this works in Rust but could we add some kind of log here such as `Server started on port {PORT}`? So instead of the `serve` function returning nothing it would at least display a message that the server is running.
```suggestion
		Ok(config) => {
		println!("INFO: Server running on port: {config.port}");
		}
```
Sorry for the bad formatting, GitHub comments section is trolling me.

Reply:
The `.await` in `server::serve(&config).await` will actually make the `serve` function start executing. If it is successful, it will start executing and block on the `.serve(service).await` methods until the server shuts down. So this match block runs _after_ that.

In that case, I can add a print to right before those methods are called.

In the case of the `Ok(config) => {`, this makes no sense. `match` is just operating on the return of `server::serve`, which returns `anyhow::Result<()>`, which is basically `Result<(), dyn Error>` from the standard library.

`Result` is an enum:
```rs
pub enum Result<T, E> {
  Ok(T),
  Err(E),
}
```

If we expand the generics, we get...

```rs
pub enum Result {
  Ok(()),
  Err(dyn Error),
}
```

In Rust, `()` is "no return", "empty value" (not null! Rust has no null), so we can't assign that `config` value there.

Reply:
I see.

We can use a tracing library later then.